src/main.rs

```diff
@@ -260,6 +273,19 @@ fn is_num(val: String) -> Result<(), String> {
     }
 }
 
+fn is_expiry(val: String) -> Result<(), String> {
+    match i64::from_str_radix(&val, 10) {
+        Ok(_) => Ok(()),
+        Err(_) => {
+            let parse_result = parse(&val);
+            match parse_result {
```

Comment:
You can simplify this a little bit and combine them

```suggestion
```

Reply:
Combined!